src/SetTypeTrait.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Phlib\Mail;
+
+/**
+ * Trait to enable the content-type to be overridden
+ *
+ * @package Phlib\Mail
+ */
+trait SetTypeTrait
```

Comment:
Not sure about the SetTypeTrait idea. I see why you're doing it.

The 'type' property should be part of the trait.